### PR TITLE
Azure disk api

### PR DESCRIPTION
**Pull request by brendandburns**
This is to update the AzureDiskApi and split it from the implementation which is caught in rebase hell...

Once this is merged, we'll get the implementation in.

@smarterclayton suggested this as a way to break the rebase hell logjam. request for a quick review.

Thanks!

### CODE REVIEW

**Comment by brendandburns** [comment]
@k8s-bot unit test this

**Comment by thockin** [comment]
/lgtm

Sorry for the hell.

**Comment by brendandburns** [comment]
(and it needs a rebase :P)

**Comment by brendandburns** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by brendandburns** [comment]
self LGTM given previous @thockin lgtm and rebase.

File: federation/docs/api-reference/extensions/v1beta1/definitions.html [hunk 1]
```diff
@@ -3032,6 +3032,13 @@ <h3 id="_v1_azurediskvolumesource">v1.AzureDiskVolumeSource</h3>
 <td class="tableblock halign-left valign-top"><p class="tableblock">boolean</p></td>
 <td class="tableblock halign-left valign-top"><p class="tableblock">false</p></td>
 </tr>
+<tr>
+<td class="tableblock halign-left valign-top"><p class="tableblock">kind</p></td>
+<td class="tableblock halign-left valign-top"><p class="tableblock">Expected values Shared: mulitple blob disks per storage account  Dedicated: single blob disk per storage account  Managed: azure managed data disk (only in managed availability set). defaults to shared</p></td>
```
Comment (rossedman) [inline federation/docs/api-reference/extensions/v1beta1/definitions.html]: Friendly bump. `mulitple` is misspelled multiple times. :laughing:

**Comment by rossedman** [comment]
@brendandburns just out of curiosity, will this address some of the PVC issues in Azure with volumes not detaching from host nodes?

**Pull request merged**